### PR TITLE
feat(generator): add vitest in generator

### PR DESCRIPTION
**Pull request by aboungnaseng-ovhcloud**
ref: MANAGER-15619

| Question         | Answer
| ---------------- | ---
| Branch?          | `master` 
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | Fix 15619
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [x] Breaking change is mentioned in relevant commits

## Description

- add vitest in generator

### CODE REVIEW

File: packages/manager/core/generator/app/templates/package.json.hbs [hunk 1]
```diff
@@ -40,10 +41,16 @@
...
+    "@testing-library/dom": "^9.3.4",
+    "@testing-library/jest-dom": "^6.4.2",
+    "@testing-library/react": "^14.2.1",
+    "@tanstack/react-query-devtools": "^5.51.21",
     "@ovh-ux/manager-vite-config": "*",
+    "@vitest/coverage-v8": "^2.1.2",
     "typescript": "^5.1.6",
-    "vite": "^5.2.13"
+    "vite": "^5.2.13",
+    "vitest": "^2.1.2"
```
Comment (tibs245) [inline packages/manager/core/generator/app/templates/package.json.hbs:52]: Think to check if the project can build with this version. I tried before and problem with the general `yarn.lock` allow to test the uApp but not the full project.

Maybe you should add 2.1.3 as hycu

File: packages/manager/core/generator/app/templates/vitest.config.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  plugins: [react()],
+  test: {
+    globals: true,
+    environment: 'jsdom',
+    setupFiles: [],
+    coverage: {
+      include: ['src'],
+      exclude: [],
+    },
+    testTimeout: 60000,
```
Comment (tibs245) [inline packages/manager/core/generator/app/templates/vitest.config.js:16]:
```suggestion
    testTimeout: 60_000,
```

May be more readable for all future project

File: packages/manager/core/generator/app/templates/package.json.hbs (hunk 1 above)
Comment (aboungnaseng-ovhcloud) [reply packages/manager/core/generator/app/templates/package.json.hbs:52]: on my side, I can launch the project, but I had to 'yarn install' at the root of the manager to be able to launch the new generated uapp project

In master, we have  2.1.2 for hycu. There is no project that use 2.1.3 at the moment

File: packages/manager/core/generator/app/templates/vitest.config.js (hunk 2 above)
Comment (aboungnaseng-ovhcloud) [reply packages/manager/core/generator/app/templates/vitest.config.js:16]: 👍

**Review by chipp972 — APPROVED** [review]

File: packages/manager/core/generator/app/conditional-templates/listing/v2/index.tsx.hbs [hunk 3]
```diff
@@ -87,7 +87,7 @@ export default function Listing() {
   }
 
   if (isLoading && !flattenData) {
-    return (<div><Loading/></div>)
+    return (<div data-testid="listing-page-spinner"><Loading/></div>)
   }
```
Comment (tibs245) [inline packages/manager/core/generator/app/conditional-templates/listing/v2/index.tsx.hbs:91]: Just for information: On ODS 18 to fix for remove <div></div> container. We need add "inline" on Loading spinner. 

I don't known if it's relevant here :)

**Review by tibs245 — APPROVED** [review]

**Pull request merged**